composer/utils/fx_utils.py

```diff
@@ -123,16 +149,106 @@ def detect_residual_pattern(gm: GraphModule):
     raise NotImplementedError('detect_residual_pattern is currently not implemented.')
 
 
-def replace_residual_with_stochastic(gm: GraphModule):
-    """Replaces residual pattern with their stoachstic equivalent.
+def _get_ancestors(node: Node) -> List[Node]:
+    ancestorNodes = []
+    while node.op != 'placeholder':
+        ancestorNodes.append(node)
+        node = node.all_input_nodes[0]
+    return ancestorNodes
+
+
+def _get_residual_path(nodeLHS: Node, nodeRHS: Node) -> Tuple[List[Node], List[Node]]:
+    """Walk backwards from nodeLHS and nodeRSH to the root and construct lists of their parents.
+
+    Arguments:
+        nodeLHS (Node): left-hand side node for a binary operator
+        nodeRHS (Node): right-hand side node for a binary operator
+
+    Returns:
+        (lhsAncestors, rhsAncesotors): Two lists of nodes containing ancestors for ``nodeLHS`` and ``nodeRHS`` with
+            their common ancestors removed.
+    """
+    lhsAncestors = _get_ancestors(nodeLHS)
```

Comment:
It goes till the root node. This is not the most efficient way but I think it's ok because graphs will have 100s of nodes rather than 1000s or millions of nodes. If it becomes a problem, we can revisit.